File: COMPETITIVE AND ALGORITHMS/SubSequences/PrintAllSubsequencesIterative.c
#include<stdio.h>
#include<math.h>
int main()
{
	int i,j;
	char a[] = "abcde";
	int n = (sizeof(a)/sizeof(a[0]))-1;
	int num = 1;
	num = pow(2, n);
	for(j=1;j<num;j++)
	{
		for(i=0;i<n;i++)
		{
			if(j>>i & 1)
			{
				printf("%c", a[i]);
			}
		}
		printf("\n");
		
	}
	return 0;
}

File: COMPETITIVE AND ALGORITHMS/SubSequences/PrintAllSubsequencesIterative.java
import java.io.*;
import java.util.*;
class PrintAllSubsequencesIterative
{
public static void main(String...args)
{
// abcd
Scanner sc = new Scanner(System.in);
char a [] = sc.nextLine().trim().toCharArray();
int n = a.length;
int num = (int)Math.pow(2, n);

for(int j = 1; j < num; j++)
{
for(int i=0;i<n;i++)
{
if( ((j >> i) & 1) == 1)
{
System.out.print(a[i]);
}
}
System.out.println();
}

}
}
